nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -177,7 +180,10 @@ bool Controller::isTrajectoryCollisionFree(
       trajectory_pub_->publish(trajectory);
       return false;
     }
-  }
+
+    // Check if we reach the goal
+    distance = nav2_util::geometry_utils::euclidean_distance(target_pose, next_pose.pose);
+  }while(distance > 1e-2 && trajectory.poses.size() < max_iter);
```

Comment:
```suggestion
  }while(distance > 1e-2 || trajectory.poses.size() < max_iter);
```

Isn't this more correct? We don't need both, but one or the other